frontend/src/pages/PipelineList.tsx

```diff
@@ -145,18 +195,34 @@ class PipelineList extends Page<{}, PipelineListState> {
       <Link
         onClick={e => e.stopPropagation()}
         className={commonCss.link}
-        to={RoutePage.PIPELINE_DETAILS.replace(':' + RouteParams.pipelineId, props.id)}
+        to={RoutePage.PIPELINE_DETAILS_NO_VERSION.replace(':' + RouteParams.pipelineId, props.id)}
       >
         {props.value}
       </Link>
     );
   };
 
-  private _selectionChanged(selectedIds: string[]): void {
-    const actions = this.props.toolbarProps.actions;
-    actions[ButtonKeys.DELETE_RUN].disabled = selectedIds.length < 1;
-    this.props.updateToolbar({ actions });
-    this.setStateSafe({ selectedIds });
+  // selection changes passed in via "selectedIds" can be
+  // (1) changes of selected pipeline ids, and will be stored in "this.state.selectedIds" or
+  // (2) changes of selected pipeline version ids, and will be stored in "selectedVersionIds" with key "pipelineId"
+  private _selectionChanged(pipelineId: string | undefined, selectedIds: string[]): void {
+    if (!!pipelineId) {
+      // Update selected pipeline version ids.
+      this.setStateSafe({
+        selectedVersionIds: { ...this.state.selectedVersionIds, ...{ [pipelineId!]: selectedIds } },
+      });
+      const actions = this.props.toolbarProps.actions;
+      actions[ButtonKeys.DELETE_RUN].disabled =
```

Comment:
This looks strange, why are we mutating props?

Reply:
My understanding is when no checkbox is checked (no pipeline or no version is selected), the delete button is gray. I kept the current behavior, and whether to change this behavior or not is open to discussion.